File: StudentOglasi/StudentOglasi.Model/PrijaveStipendija.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace StudentOglasi.Model
{
    public class PrijaveStipendija
    {
        public int StudentId { get; set; }

        public int StipendijaId { get; set; }

        public string Dokumentacija { get; set; } = null!;

        public string Cv { get; set; } = null!;

        public decimal ProsjekOcjena { get; set; }

        public int StatusId { get; set; }

        public virtual StatusPrijave Status { get; set; } = null!;

        public virtual Stipendije Stipendija { get; set; } = null!;

        public virtual Studenti Student { get; set; } = null!;
    }
}

File: StudentOglasi/UI/studentoglasi_admin/lib/models/PrijaveStipendija/prijave_stipendija.dart

import 'package:json_annotation/json_annotation.dart';
import 'package:studentoglasi_admin/models/StatusPrijave/statusprijave.dart';
import 'package:studentoglasi_admin/models/Stipendija/stipendija.dart';
import 'package:studentoglasi_admin/models/Student/student.dart';

part 'prijave_stipendija.g.dart';

@JsonSerializable()
class PrijaveStipendija {
  int? studentId;
  int? stipendijaId;
  String? dokumentacija;
  String? cv;
  double? prosjekOcjena;
  StatusPrijave? status;
  Stipendije? stipendija;
  Student? student;


  PrijaveStipendija(this.studentId, this.stipendijaId, this.dokumentacija,this.cv,this.prosjekOcjena,this.status,this.stipendija,this.student);
  factory PrijaveStipendija.fromJson(Map<String, dynamic> json) => _$PrijaveStipendijaFromJson(json);

  Map<String, dynamic> toJson() => _$PrijaveStipendijaToJson(this);
}
